lib/id3c/utils.py

```diff
@@ -14,3 +16,46 @@ def wrap(function):
         function.__doc__ = function.__doc__.format_map(kwargs)
         return function
     return wrap
+
+
+def getattrpath(value: Any, attrpath: Union[str, Sequence[str]]) -> Any:
+    """
+    Get a nested named attribute, described by *attrpath*, from the
+    object *value*, or return None if one of the attributes in the path
+    doesn't exist.
+
+    *attrpath* may be a dotted-string like ``a.b.c`` or a sequence
+    (list or tuple) like ``("a", "b", "c")``.
+
+    >>> class Namespace:
+    ...     def __init__(self):
+    ...         self.__dict__ = {}
+    ...     def __repr__(self):
+    ...         return repr(self.__dict__)
+
+    >>> obj = Namespace()
+    >>> obj.a = Namespace()
+    >>> obj.a.b = Namespace()
+    >>> obj.a.b.c = 42
+    >>> getattrpath(obj, "a")
+    {'b': {'c': 42}}
+    >>> getattrpath(obj, "a.b.c")
+    42
+    >>> getattrpath(obj, "a.b.x")
+    >>> getattrpath(obj, "a.x")
+    >>> getattrpath(obj, "x")
+    >>> getattrpath(obj, ("a", "b", "c"))
+    42
```

Comment:
Those would make great unit tests. Do we have unit tests for utility functions like these?

Reply:
These are part of the [doctests](https://docs.python.org/3/library/doctest.html) that get run with pytest.

Reply:
Ah, nice. Thanks for the link.